test/TestUtilities/Test.Utility/Signing/TrustedTestCert.cs

```diff
@@ -84,6 +85,31 @@ private void AddCertificateToStore()
             _store = new X509Store(StoreName, StoreLocation);
             _store.Open(OpenFlags.ReadWrite);
             _store.Add(TrustedCert);
+
+            //Add wait for Linux, as https://github.com/dotnet/runtime/issues/32608
+            //Windows has a live-synchronized model, and on Linux, there is a filesystem/rescan delay problem.
+            //For performance reasons, dotnet/runtime only check to see if the store has been modified once a second.
+            if (RuntimeEnvironmentHelper.IsLinux)
+            {
+                var MaxTries = 30;
+
+                for (var i = 0; i < MaxTries; i++)
```

Comment:
Can you put a `Thread.Sleep(1500)` or `2000` right before this loop? We know for sure that we at least need to wait a second, so we should lead with that. Otherwise there's a pointless chain build that we already know will fail.

Reply:
Thanks for review! The problem is the test time on Linux has extended from about 45mins to 90mins.Is that acceptable? Or shall I log the time to see if the time of packaging tests doubled?

Reply:
I truned the "verbosity -detailed" on and found a test runs 10-40 mins
ExecuteCommandAsync_WithAmbiguousMatch_ThrowsAsync

Raised an issue https://github.com/NuGet/Home/issues/9315 and I will investigate it.